compiler+runtime/src/cpp/jank/read/lex.cpp

```diff
@@ -260,11 +266,53 @@ namespace jank::read
       return none;
     }
 
-    static native_bool is_symbol_char(char const c)
+    static result<codepoint, error>
+    convert_to_codepoint(native_persistent_string_view const sv, size_t const pos)
+    {
+      std::mbstate_t state{};
+      wchar_t wc{};
+      auto const len{ std::mbrtowc(&wc, sv.data(), sv.size(), &state) };
+
+      if(len == static_cast<size_t>(-1))
+      {
+        return err(error{ pos, "Unfinished Character" });
+      }
+      else if(len == static_cast<size_t>(-2))
+      {
+        return err(error{ pos, "Invalid character" });
+      }
+      return ok(codepoint{ static_cast<char32_t>(wc), len });
+    }
+
+    static native_bool is_utf8_char(char32_t const c)
+    {
+      if(c <= 0x7FF)
+      {
+        return true;
+      }
+      else if(c <= 0xFFFF)
+      {
+        return c < 0xD800 || c > 0xDFFF;
+      }
+      else if(c <= 0x10FFFF)
+      {
+        return true;
+      }
```

Comment:
This is good code, but these values are magic! This is the perfect sort of fn which needs a comment inside which at the very least links to the resource used to do this logic and, at best, also describes how it's working.